In [1]:
import pandas as pd

## Usando features existentes

Para ahorrar tiempo, en una primera instancia vamos a usar features ya extraídas desde la web api de Spotify que están disponibles en kaggle.

En particular usaremos [Spotify Dataset](https://www.kaggle.com/datasets/vatsalmavani/spotify-dataset) (171k canciones) y [Spotify 1.2M+ Songs](https://www.kaggle.com/datasets/rodolfofigueroa/spotify-12m-songs) (1.2M canciones).

In [2]:
df_playlists = pd.read_csv('data/1m.csv')
df_audio_features = pd.read_csv('data/data.csv') #171k
df_audio_features2 = pd.read_csv('data/tracks_features.csv') #1.2m

df_playlists['track_uri'] = df_playlists['track_uri'].str.replace('spotify:track:', '')
df_audio_features['track_uri'] = df_audio_features['id']
df_audio_features2['track_uri'] = df_audio_features2['id']
df_audio_features = pd.concat([df_audio_features, df_audio_features2], ignore_index=True)
df_audio_features = df_audio_features.drop_duplicates(subset='track_uri')

In [3]:
merged = pd.merge(df_playlists, df_audio_features, on='track_uri', how='inner')


In [6]:
merged['playlist_id'] = merged['Unnamed: 0']
merged.drop(['Unnamed: 0'], axis=1, inplace=True)

In [7]:
merged.to_csv('data/merged.csv')

In [8]:
len(merged)/len(df_playlists.track_uri) 

0.08324256992466048

Usando features preexistentes, nos quedamos con un 8.32% de las canciones en el dataset original.

In [44]:
df = pd.read_csv('data/merged.csv')
recomendations = pd.read_csv('data/1m.csv')
recomendations['track_uri'] = recomendations['track_uri'].apply(lambda x: x.split(':')[2])
filtered_recomendations = recomendations[recomendations['track_uri'].isin(df['track_uri'])]
filtered_recomendations.to_csv('data/filtered_recomendations.csv')

In [43]:
len(filtered_recomendations['Unnamed: 0'].unique())

1000